In [1]:
import numpy as np
import pandas as pd
import sys
sys.path.append('..')
from models.feature_extraction.feature_extraction import Data

In [2]:
data = Data()
data.LL_data = pd.read_csv('../LL_data.csv', index_col=0)
data.SL_data = pd.read_csv('../SL_data.csv', index_col=0)
data.matches = pd.read_csv('../matches.csv', parse_dates=['Date','Open'], index_col=0)
data.match_data = pd.read_csv('../match_data.csv', parse_dates=['Date'],index_col=0)

In [6]:
productivity = pd.DataFrame(index=data.matches.index)
for i in [1,2,5,10]:
    productivity['HID_goals_scored_to_match_#_m#'+str(i)] = data.matches.apply(lambda row: data.goals_to_match_ratio(row.HID,number_of_matches=i,MatchID=row.name,goal_type='scored'), axis=1)
    productivity['HID_goals_conceded_to_match_#_m#'+str(i)] = data.matches.apply(lambda row: data.goals_to_match_ratio(row.HID,number_of_matches=i,MatchID=row.name,goal_type='conceded'), axis=1) 
    productivity['AID_goals_scored_to_match_#_m#'+str(i)] = data.matches.apply(lambda row: data.goals_to_match_ratio(row.AID,number_of_matches=i,MatchID=row.name,goal_type='scored'), axis=1)
    productivity['AID_goals_conceded_to_match_#_m#'+str(i)] = data.matches.apply(lambda row: data.goals_to_match_ratio(row.AID,number_of_matches=i,MatchID=row.name,goal_type='conceded'), axis=1)                                                                            

In [9]:
productivity[['A','H']] = data.matches[['A','H']]

In [17]:
productivity[[col for col in productivity.columns if 'HID' in col and 'scored' in col] + ['H']].corr()

HID_goals_scored_to_match_#_m#1  \
HID_goals_scored_to_match_#_m#1                          1.000000   
HID_goals_scored_to_match_#_m#2                          0.702059   
HID_goals_scored_to_match_#_m#5                          0.482976   
HID_goals_scored_to_match_#_m#10                         0.365118   
H                                                        0.034722   

                                  HID_goals_scored_to_match_#_m#2  \
HID_goals_scored_to_match_#_m#1                          0.702059   
HID_goals_scored_to_match_#_m#2                          1.000000   
HID_goals_scored_to_match_#_m#5                          0.683104   
HID_goals_scored_to_match_#_m#10                         0.525377   
H                                                        0.054674   

                                  HID_goals_scored_to_match_#_m#5  \
HID_goals_scored_to_match_#_m#1                          0.482976   
HID_goals_scored_to_match_#_m#2                          0.683104   
HID_goals_scored_to_match_#_m#5                          1.000000   
HID_goals_scored_to_match_#_m#10                         0.793073   
H                                                        0.080106   

                                  HID_goals_scored_to_match_#_m#10         H  
HID_goals_scored_to_match_#_m#1                           0.365118  0.034722  
HID_goals_scored_to_match_#_m#2                           0.525377  0.054674  
HID_goals_scored_to_match_#_m#5                           0.793073  0.080106  
HID_goals_scored_to_match_#_m#10                          1.000000  0.088218  
H                                                         0.088218  1.000000

In [19]:
productivity[[col for col in productivity.columns if 'AID' in col and 'scored' in col] + ['A']].corr()

AID_goals_scored_to_match_#_m#1  \
AID_goals_scored_to_match_#_m#1                          1.000000   
AID_goals_scored_to_match_#_m#2                          0.740388   
AID_goals_scored_to_match_#_m#5                          0.511930   
AID_goals_scored_to_match_#_m#10                         0.387473   
A                                                        0.047612   

                                  AID_goals_scored_to_match_#_m#2  \
AID_goals_scored_to_match_#_m#1                          0.740388   
AID_goals_scored_to_match_#_m#2                          1.000000   
AID_goals_scored_to_match_#_m#5                          0.691314   
AID_goals_scored_to_match_#_m#10                         0.530911   
A                                                        0.068906   

                                  AID_goals_scored_to_match_#_m#5  \
AID_goals_scored_to_match_#_m#1                          0.511930   
AID_goals_scored_to_match_#_m#2                          0.691314   
AID_goals_scored_to_match_#_m#5                          1.000000   
AID_goals_scored_to_match_#_m#10                         0.795828   
A                                                        0.093932   

                                  AID_goals_scored_to_match_#_m#10         A  
AID_goals_scored_to_match_#_m#1                           0.387473  0.047612  
AID_goals_scored_to_match_#_m#2                           0.530911  0.068906  
AID_goals_scored_to_match_#_m#5                           0.795828  0.093932  
AID_goals_scored_to_match_#_m#10                          1.000000  0.116055  
A                                                         0.116055  1.000000

Jak je vidět, tak korelace je největší pro vstřelené góly za posledních 10 zápasů a to $0.088218$ pro `HID` a $0.116055$ pro `AID`

In [21]:
productivity[[col for col in productivity.columns if 'HID' in col and 'conceded' in col] + ['A']].corr()

HID_goals_conceded_to_match_#_m#1  \
HID_goals_conceded_to_match_#_m#1                            1.000000   
HID_goals_conceded_to_match_#_m#2                            0.746214   
HID_goals_conceded_to_match_#_m#5                            0.502454   
HID_goals_conceded_to_match_#_m#10                           0.373911   
A                                                            0.059901   

                                    HID_goals_conceded_to_match_#_m#2  \
HID_goals_conceded_to_match_#_m#1                            0.746214   
HID_goals_conceded_to_match_#_m#2                            1.000000   
HID_goals_conceded_to_match_#_m#5                            0.683079   
HID_goals_conceded_to_match_#_m#10                           0.511887   
A                                                            0.074280   

                                    HID_goals_conceded_to_match_#_m#5  \
HID_goals_conceded_to_match_#_m#1                            0.502454   
HID_goals_conceded_to_match_#_m#2                            0.683079   
HID_goals_conceded_to_match_#_m#5                            1.000000   
HID_goals_conceded_to_match_#_m#10                           0.782928   
A                                                            0.097700   

                                    HID_goals_conceded_to_match_#_m#10  \
HID_goals_conceded_to_match_#_m#1                             0.373911   
HID_goals_conceded_to_match_#_m#2                             0.511887   
HID_goals_conceded_to_match_#_m#5                             0.782928   
HID_goals_conceded_to_match_#_m#10                            1.000000   
A                                                             0.106226   

                                           A  
HID_goals_conceded_to_match_#_m#1   0.059901  
HID_goals_conceded_to_match_#_m#2   0.074280  
HID_goals_conceded_to_match_#_m#5   0.097700  
HID_goals_conceded_to_match_#_m#10  0.106226  
A                                   1.000000

In [20]:
productivity[[col for col in productivity.columns if 'AID' in col and 'conceded' in col] + ['H']].corr()

AID_goals_conceded_to_match_#_m#1  \
AID_goals_conceded_to_match_#_m#1                            1.000000   
AID_goals_conceded_to_match_#_m#2                            0.707825   
AID_goals_conceded_to_match_#_m#5                            0.478686   
AID_goals_conceded_to_match_#_m#10                           0.351765   
H                                                            0.044612   

                                    AID_goals_conceded_to_match_#_m#2  \
AID_goals_conceded_to_match_#_m#1                            0.707825   
AID_goals_conceded_to_match_#_m#2                            1.000000   
AID_goals_conceded_to_match_#_m#5                            0.679944   
AID_goals_conceded_to_match_#_m#10                           0.510888   
H                                                            0.057627   

                                    AID_goals_conceded_to_match_#_m#5  \
AID_goals_conceded_to_match_#_m#1                            0.478686   
AID_goals_conceded_to_match_#_m#2                            0.679944   
AID_goals_conceded_to_match_#_m#5                            1.000000   
AID_goals_conceded_to_match_#_m#10                           0.779885   
H                                                            0.071921   

                                    AID_goals_conceded_to_match_#_m#10  \
AID_goals_conceded_to_match_#_m#1                             0.351765   
AID_goals_conceded_to_match_#_m#2                             0.510888   
AID_goals_conceded_to_match_#_m#5                             0.779885   
AID_goals_conceded_to_match_#_m#10                            1.000000   
H                                                             0.079180   

                                           H  
AID_goals_conceded_to_match_#_m#1   0.044612  
AID_goals_conceded_to_match_#_m#2   0.057627  
AID_goals_conceded_to_match_#_m#5   0.071921  
AID_goals_conceded_to_match_#_m#10  0.079180  
H                                   1.000000

Na inkasované góly za posledních 10 zápasů opět celkem vysoká korelace s prohrou (nejvyšší z # zápasů) a to $0.106226$ pro `HID` a $0.079180$ pro `AID`